hail/python/hailtop/batch/genetics/regenie/regenie.py

```diff
@@ -256,60 +254,71 @@ def prepare_jobs(batch, args: BatchArgs, step1_args: Namespace, step2_args: Name
 
     batch = Batch(backend=backend, name='regenie')
 
-    j2 = prepare_jobs(batch, batch_args, step1_args, step2_args)
-
-    j2out = splitext(basename(step2_args.out))[0]
-    if args.outdir[-1] != "/":
-        outpath = f"{args.outdir}/{j2out}"
-    else:
-        outpath = f"{args.outdir}{j2out}"
-
-    print(f"Output path set as: {outpath}")
-    batch.write_output(j2.output, outpath)
-    batch.run(**run_opts)
+    j2 = prepare_jobs(batch, step1_args, step1_batch_args, step2_args, step2_batch_args)
+    print(f"Will write output to: {step2_args.out}")
+    batch.write_output(j2.output, step2_args.out)
+    return batch.run(**run_opts)
 
 
-def parse_input_args(args: list):
+def parse_input_args(input_args: list):
     parser = ArgumentParser()
     parser.add_argument('--local', required=False, action="store_true")
     parser.add_argument('--demo', required=False, action="store_true")
-    parser.add_argument('--outdir', required=True)
-    # FIXME: replace with per-step resources
-    parser.add_argument('--cores', required=False, default=2)
-    parser.add_argument('--memory', required=False, default="7Gi")
-    parser.add_argument('--storage', required=False, default="1Gi")
     parser.add_argument('--step1', required=False)
     parser.add_argument('--step2', required=False)
 
-    return parser.parse_args(args)
+    args = parser.parse_known_args(input_args)[0]
+    backend_parser = ArgumentParser(argument_default=SUPPRESS)
+    run_parser = ArgumentParser(argument_default=SUPPRESS)
+    if args.local:
+        backend_parser.add_argument('--tmp_dir', required=False)
+        backend_parser.add_argument('--gsa_key_file', required=False)
+        backend_parser.add_argument('--extra_docker_run_flags', required=False)
+
+        run_parser.add_argument('--dry_run', required=False, action="store_true")
+        run_parser.add_argument('--verbose', required=False, action="store_true")
+        run_parser.add_argument('--delete_scratch_on_exit', required=False, action="store_true")
+    else:
+        backend_parser.add_argument('--billing_project', required=False)
+        backend_parser.add_argument('--bucket', required=False)
+
+        run_parser.add_argument('--dry_run', required=False, action="store_true")
+        run_parser.add_argument('--verbose', required=False, action="store_true")
+        run_parser.add_argument('--delete_scratch_on_exit', required=False, action="store_true")
+        run_parser.add_argument('--wait', required=False, action="store_true")
+        run_parser.add_argument('--open', required=False, action="store_true")
+        run_parser.add_argument('--disable_progress_bar', required=False, action="store_true")
+        run_parser.add_argument('--callback', required=False)
+
+    backend_args = backend_parser.parse_known_args(input_args)[0]
+    run_args = run_parser.parse_known_args(input_args)[0]
+
+    return {"args": args, "backend_opts": vars(backend_args), "run_opts": vars(run_args)}
```

Comment:
Using a different Argument Parser instance does lead to elegant code; however, the fact that the help message does not documentation the mere existence of these other arguments seems to doom the approach. Moreover, since these arguments are not REGENIE-specific, I think we ought to include a help string.